lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/system/MongoUserNotificationDao.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.lumeer.storage.mongodb.dao.system;
+
+import static io.lumeer.storage.mongodb.util.MongoFilters.idFilter;
+
+import io.lumeer.api.model.UserNotification;
+import io.lumeer.storage.api.dao.UserNotificationDao;
+import io.lumeer.storage.api.exception.StorageException;
+import io.lumeer.storage.mongodb.codecs.UserNotificationCodec;
+
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.FindOneAndReplaceOptions;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.Sorts;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * @author <a href="mailto:marvenec@gmail.com">Martin Večeřa</a>
+ */
+@ApplicationScoped
+public class MongoUserNotificationDao extends SystemScopedDao implements UserNotificationDao {
+
+   public static final String COLLECTION_NAME = "userNotifications";
+
+   @PostConstruct
+   public void initDb() {
+      createUserNotificationsRepository();
+   }
+
+   @Override
+   public List<UserNotification> getRecentNotifications(final String userId) {
+      Bson idFilter = Filters.eq(UserNotificationCodec.USER_ID, userId);
+
+      return databaseCollection()
+            .find(idFilter)
+            .limit(500)
```

Comment:
500? Are you sure?

Reply:
Let's be generous ;-) What it is, like 10kB?